apps/molgenis-components/src/components/forms/EditModal.vue

```diff
@@ -295,75 +293,21 @@ export default {
     this.tableMetaData = await this.client.fetchTableMetaData(this.tableName);
 
     if (this.pkey) {
-      this.rowData = await this.fetchRowData();
+      const rowData = await this.fetchRowData();
 
       if (this.clone) {
-        // in case of clone, remove the key columns from the row data
-        const keyColumnsNames = this.tableMetaData?.columns
-          ?.filter((column: IColumn) => column.key === 1)
-          .map((column: IColumn) => column.name);
-
-        this.rowData = filterObject(
-          this.rowData,
-          (key) => !keyColumnsNames?.includes(key)
-        );
+        this.rowData = removeKeyColumns(this.tableMetaData, rowData);
```

Comment:
suggest to keep the code comment here , it says something about the why

Reply:
The why is already in the if clause, the what is in the function name. The comment would be double bookkeeping.